File: python/codeeval/010_mth_to_last.py
import sys

def m(test):
    l = test.split(' ')
    i = int(l.pop())
    idx = len(l) - i
    if idx >= 0:
        v = l[idx]
        if (v is not None):
            print(v)

with open(sys.argv[1], 'r') as test_cases:
    for test in test_cases:
        m(test)

File: ruby/codeeval/010_mth_to_last.rb
def m(line)
    puts(line)
    r = line.strip.split(' ')
    i = r.pop.to_i
    id = r.length - i
    l = id >= 0 ? r[id] : nil
    puts(l) if l
end

File.open(ARGV[0]).each_line do |line|
    m(line)
end
